action.yml

```diff
@@ -0,0 +1,69 @@
+name: Branch commit and pull-request
+description: GitHub Action to create a branch, commit and a pull-request as the github-actions user
+
+inputs:
+  branch:
+    type: string
+    required: true
+  title:
+    type: boolean
+    required: true
+  description:
+    type: string
+    required: false
+    default: ''
+
+runs:
+  using: composite
+  steps:
+
+    - name: Validate branch
+      shell: bash
+      env:
+        BRANCH: ${{ inputs.branch }}
+      run: |
+        if [[ "$BRANCH" =~ [^a-zA-Z0-9\./_\-] ]] || [ -z "$BRANCH" ]; then
+          echo "Invalid branch name"
+          exit 1
+        fi
+        if ! [[ -z $(git ls-remote --heads origin $BRANCH) ]]; then
+          echo "Branch $BRANCH already exists"
+          exit 1
+        fi
+      
+    - name: Branch commit and pull-request
+      shell: bash
+      # Add string inputs to memory instead of using string substitution in shell script
+      # https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable
+      env:
+        BRANCH: ${{ inputs.branch }}
+        TITLE: ${{ inputs.title }}
+        DESCRIPTION: ${{ inputs.description }}
+      run: |
+        # Escape double quotes '"' => '\"'
+        TITLE=${TITLE//\"/\\\"}
+        DESCRIPTION=${DESCRIPTION//\"/\\\"}
+        BASE_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+        # Run git commit, push and create pull-request as 'github-actions' user
+        git config --local user.name "github-actions"
+        git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
```

Comment:
Nope, I copied it from an example somewhere else long ago.  I've removed the `41898282+` prefix from the email, I guess we'll shortly see if it still work without it

Reply:
I think it might actually be necessary - https://github.community/t/github-actions-bot-email-address/17204/6
> [https://api.github.com/users/github-actions[bot]](https://api.github.com/users/github-actions%5Bbot%5D) seems to be the API entry for the github-actions bot, which also contains the ID 41898282.

So, sorry to be a pain but can you please put it back?

Reply:
Added back and added comment